framework/src/controller/bus.ts

```diff
@@ -311,13 +311,15 @@ export class Bus {
 		this._emitter.emit(eventName, notification);
 
 		// Communicate through unix socket
-		try {
-			this._ipcServer.pubSocket.send(notification);
-		} catch (error) {
-			this.logger.debug(
-				{ err: error as Error },
-				`Failed to publish event: ${eventName} to ipc server.`,
-			);
+		if (this.config.rpc.enable) {
+			try {
+				this._ipcServer.pubSocket.send(notification);
+			} catch (error) {
+				this.logger.debug(
+					{ err: error as Error },
+					`Failed to publish event: ${eventName} to ipc server.`,
+				);
+			}
```

Comment:
```suggestion
		if (this.config.rpc.enable) {
			try {
				this._ipcServer.pubSocket.send(notification);
			} catch (error) {
				this.logger.debug(
					{ err: error as Error },
					`Failed to publish event: ${eventName} to ipc server.`,
				);
			}
			
			if (this.config.rpc.enable && this.config.rpc.mode === 'ws') {
				try {
					this._wsServer.broadcast(JSON.stringify(notification));
				} catch (error) {
					this.logger.debug(
						{ err: error as Error },
						`Failed to publish event: ${eventName} to ws server.`,
					);
				}
			}
```

Reply:
I think if we can club the conditional block starting at `L325` here like the suggestion above, it saves one additional `if` comparison. Feel free to ignore though since it's not a significant difference.